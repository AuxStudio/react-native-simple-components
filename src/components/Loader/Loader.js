import React from 'react';
import PropTypes from 'prop-types';
import { View, ViewPropTypes } from 'react-native';
import { AnimateOpacity, AnimateTranslateX } from 'react-native-simple-animators';

import styleConstants from '../../styleConstants';

const propTypes = {
  width: PropTypes.number,
  height: PropTypes.number,
  color: PropTypes.string, // color of the loader
  duration: PropTypes.number, // duration of animation in ms
  style: ViewPropTypes.style,
  containerStyle: ViewPropTypes.style,
  wrapperStyle: ViewPropTypes.style,
};

const defaultProps = {
  width: 100,
  height: 5,
  color: styleConstants.colors.primary,
  duration: 2000,
};

const Loader = ({ width, height, color, duration, style, containerStyle, wrapperStyle }) => {
  return (
    <AnimateOpacity initialValue={0} finalValue={1} shouldAnimateIn style={wrapperStyle}>
      <AnimateTranslateX
        initialValue={width * -1}
        finalValue={styleConstants.dimensions.window.width}
        shouldAnimateIn
        shouldRepeat
        shouldLoop
        duration={duration}
        style={containerStyle}
      >
        <View
          style={[
            {
              width,
              height,
              backgroundColor: color,
            },
            style,
          ]}
        />
      </AnimateTranslateX>
    </AnimateOpacity>
  );
};

Loader.propTypes = propTypes;
Loader.defaultProps = defaultProps;

export default Loader;
