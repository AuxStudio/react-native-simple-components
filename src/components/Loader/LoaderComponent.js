import React from "react";
import { View, StyleSheet } from "react-native";
import PropTypes from "prop-types";

import styleConstants from "../../assets/styleConstants";

import {
    AnimateOpacity,
    AnimateTranslateX,
} from "react-native-simple-animators";

export default (LoaderComponent = props => {
    /*
        static get propTypes() {
            return {
                color: PropTypes.string,
                duration: PropTypes.number,
                // style: PropTypes.node,
            };
        }
    */

    return (
        <AnimateOpacity initialValue={0} finalValue={1} shouldAnimateIn>
            <AnimateTranslateX
                initialValue={-100 /* width of loader */}
                finalValue={styleConstants.windowWidth}
                shouldAnimateIn
                shouldRepeat
                shouldLoop
                duration={props.duration ? props.duration : 2000}>
                <View
                    style={[
                        styles.container,
                        {
                            backgroundColor: props.color
                                ? props.color
                                : styleConstants.secondary,
                        },
                        props.style,
                    ]}
                />
            </AnimateTranslateX>
        </AnimateOpacity>
    );
});

const styles = StyleSheet.create({
    container: {
        width: 100,
        height: 5,
    },
});
