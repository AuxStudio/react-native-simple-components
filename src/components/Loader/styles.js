import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  wrapper: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    height: 5,
  },
  container: {
    width: 100,
    height: 5,
  },
});

export default styles;
