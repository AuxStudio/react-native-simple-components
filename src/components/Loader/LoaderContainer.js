import React from "react";
import PropTypes from "prop-types";
import { View, StyleSheet } from "react-native";
import { connect } from "react-redux";

import LoaderComponent from "./LoaderComponent";

const styles = StyleSheet.create({
    container: {
        position: "absolute",
        bottom: 0,
        left: 0,
        right: 0,
        height: 5,
    },
});

export class Loader extends React.Component {
    static get propTypes() {
        return {
            loading: PropTypes.bool,
            forceLoading: PropTypes.bool,
            color: PropTypes.string,
        };
    }

    render() {
        const loader = (this.props.loading || this.props.forceLoading) && (
            <LoaderComponent color={this.props.color} />
        );

        return <View style={styles.container}>{loader}</View>;
    }
}

function mapStateToProps(state) {
    return {
        loading: state.main.appState.loading,
    };
}

export default connect(mapStateToProps)(Loader);
