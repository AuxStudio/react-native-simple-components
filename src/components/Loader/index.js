import LoaderContainer from './LoaderContainer';

export default Loader = LoaderContainer;