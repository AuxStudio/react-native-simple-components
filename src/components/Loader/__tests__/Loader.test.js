/* eslint-disable import/first */
import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('react-native-simple-animators', () => ({
  AnimateOpacity: 'AnimateOpacity',
  AnimateTranslateX: 'AnimateTranslateX',
}));

import Loader from '../';

it('renders a Loader', () => {
  expect(
    renderer.create(
      <Loader
        width={200}
        height={10}
        color="red"
        duration={500}
        style={{ backgroundColor: 'blue' }}
        containerStyle={{ backgroundColor: 'green' }}
        wrapperStyle={{ backgroundColor: 'purple' }}
      />,
    ),
  ).toMatchSnapshot();
});

it('renders a Loader with no props', () => {
  expect(renderer.create(<Loader />)).toMatchSnapshot();
});
/* eslint-enable */
