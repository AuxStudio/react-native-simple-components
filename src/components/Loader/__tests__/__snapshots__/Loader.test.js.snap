// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a Loader 1`] = `
<AnimateOpacity
  finalValue={1}
  initialValue={0}
  shouldAnimateIn={true}
  style={
    Object {
      "backgroundColor": "purple",
    }
  }
>
  <AnimateTranslateX
    duration={500}
    finalValue={750}
    initialValue={-200}
    shouldAnimateIn={true}
    shouldLoop={true}
    shouldRepeat={true}
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "red",
            "height": 10,
            "width": 200,
          },
          Object {
            "backgroundColor": "blue",
          },
        ]
      }
    />
  </AnimateTranslateX>
</AnimateOpacity>
`;

exports[`renders a Loader with no props 1`] = `
<AnimateOpacity
  finalValue={1}
  initialValue={0}
  shouldAnimateIn={true}
>
  <AnimateTranslateX
    duration={2000}
    finalValue={750}
    initialValue={-100}
    shouldAnimateIn={true}
    shouldLoop={true}
    shouldRepeat={true}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#3D84A8",
            "height": 5,
            "width": 100,
          },
          undefined,
        ]
      }
    />
  </AnimateTranslateX>
</AnimateOpacity>
`;
